Source/Shaders/ModelExperimental/GeometryStageVS.glsl

```diff
@@ -4,7 +4,14 @@ void geometryStage(inout ProcessedAttributes attributes, mat4 modelView, mat3 no
     vec3 positionMC = attributes.positionMC;
     v_positionMC = positionMC;
     v_positionEC = (modelView * vec4(positionMC, 1.0)).xyz;
+
+    #ifdef USE_2D_POSITIONS
```

Comment:
The lighting is currently incorrect so I think this is a mistake on my end. I'll have to see if using it for v_positionEC will change things.

Reply:
Definitely curious to see the difference here since I'm not sure which looks better: as it stands it's like you're lighting in 3D space and projecting the results. The other way would be lighting the 2D scene after projecting.

The other thing to consider is how custom shaders should react to 2D mode

Reply:
Maybe other users could weigh in, but personally I'm thinking... say I have a custom shader that shades models based on height. If I go into 2D mode, wouldn't I want my custom shader to be the same?

Reply:
Indeed. If the original model coordinates was an ENU frame, then `positionMC.z` is the height. But then based on how the projection works, that height value is passed through unaltered, but it winds up in `position2D.x`, which would probably be surprising. So yeah, perhaps using the 3D coordinates in custom shader is indeed best here.